gps_common/src/utm_odometry_node.cpp

```diff
@@ -11,166 +11,194 @@
+      priv_node.param<bool>("use_fixed_origin", use_fixed_origin, false);
+
+      // if use_fixed_origin is false, wait for a /inspva message
+      // and use it as the origin
+      if (!use_fixed_origin) {
+        auto ins = ros::topic::waitForMessage<novatel_gps_msgs::Inspva>("/inspva", node);
+        double northing, easting;
+        std::string zone;
+        LLtoUTM(ins->latitude, ins->longitude, northing, easting, zone);
+        origin.pose.position.x = easting;
+        origin.pose.position.y = northing;
+        origin.pose.position.z = ins->height;
+        origin.pose.orientation = tf::createQuaternionMsgFromRollPitchYaw(
+          ins->roll * DEG2RAD,
+          ins->pitch * DEG2RAD,
+          (90.0 - ins->azimuth) * DEG2RAD);
+      } else {
+        // TODO more than parameters, a service call would be better
+        // so that we could change the origin at runtime
+        priv_node.param<double>("origin_x", origin.pose.position.x, 0);
+        priv_node.param<double>("origin_y", origin.pose.position.y, 0);
+        priv_node.param<double>("origin_z", origin.pose.position.z, 0);
+        // TODO using euler angles might be easier
+        priv_node.param<double>("origin_ox", origin.pose.orientation.x, 0);
+        priv_node.param<double>("origin_oy", origin.pose.orientation.y, 0);
+        priv_node.param<double>("origin_oz", origin.pose.orientation.z, 0);
+        priv_node.param<double>("origin_ow", origin.pose.orientation.w, 1);
+      }
+
+      if (ignore_z)
+        origin.pose.position.z = 0;
+
+      origin.header.frame_id = world_frame_id;
+      origin.header.stamp = ros::Time::now();
+
+      PublishOrigin(origin);
+    }
+
+    void PublishOrigin(const geometry_msgs::PoseStamped& origin) {
+      origin_pub = node.advertise<geometry_msgs::PoseStamped>("/origin", 1, true);
+      origin_pub.publish(origin);
+
+      const Eigen::Vector3d translation(origin.pose.position.x,
+        origin.pose.position.y,
+        origin.pose.position.z);
+      const Eigen::Quaterniond rotation(origin.pose.orientation.w,
+        origin.pose.orientation.x,
+        origin.pose.orientation.y,
+        origin.pose.orientation.z);
+        world_to_local_frame = std::make_pair(rotation * -translation, rotation);
+    }
+
+    void PublishOdomTF(const nav_msgs::Odometry& odom) {
+      odom_tf.header.stamp = odom.header.stamp;
+      odom_tf.transform.translation.x = odom.pose.pose.position.x;
+      odom_tf.transform.translation.y = odom.pose.pose.position.y;
+      odom_tf.transform.translation.z = odom.pose.pose.position.z;
+
+      odom_tf.transform.rotation.x = odom.pose.pose.orientation.x;
+      odom_tf.transform.rotation.y = odom.pose.pose.orientation.y;
+      odom_tf.transform.rotation.z = odom.pose.pose.orientation.z;
+      odom_tf.transform.rotation.w = odom.pose.pose.orientation.w;
+
+      odom_tf_broadcaster.sendTransform(odom_tf);
+    }
+
+    void INSPVACallBack(const novatel_gps_msgs::InspvaConstPtr& ins) {
```

Comment:
member functions are camelCase according to google style guide.

- [ ] change to "inspvaCallBack(const novatel_gps_msgs::InspvaConstPtr& ins)"

Reply:
I tried to follow this [https://google.github.io/styleguide/cppguide.html#Function_Names]()

Reply:
you are totally right, sorry I was thinking of ROS style guide. we can keep it the way it is.

Reply:
I'm not quite sure which guidelines we should follow so as much as possible I try to follow the google one but I'm OK using whatever we choose.